spec/controllers/assignment_invitations_controller_spec.rb

```diff
@@ -68,6 +68,30 @@
       end
     end
 
+    context 'github repository with the same name already exists' do
+      let(:assignment_repo) { AssignmentRepo.new(assignment: assignment, user: user) }
+
+      before do
+        assignment_repo.create_github_repository
+      end
+
+      it 'creates a new assignment repo' do
+        patch :accept_invitation, id: invitation.key
+        expect(user.assignment_repos.count).to eql(1)
+      end
+
+      it 'new repository name has expected suffix' do
+        patch :accept_invitation, id: invitation.key
+
+        expect(WebMock).to have_requested(:post, github_url("/organizations/#{organization.github_id}/repos"))
+          .with(body: /^.*#{assignment_repo.repo_name}-1.*$/)
+      end
+
+      after do
+        assignment_repo.destroy_github_repository
+      end
+    end
+
```

Comment:
This test was kind of confusing to me just because of how it was setup.

WDYT about something like this?

```ruby
context 'github repository with the name same already exists' do
  
  before do
    assignment_repo = AssignmentRepo.new(assignment: assignment, user: user)
    @original_repository = organization.github_client.repository(assignment_repo.github_repo_id)
    
    # use delete instead of destroy to skip active record callbacks
    assignment_repo.delete

    patch :accept_invitation, id: invitation.key
  end
  
  after do
    organization.github_client.delete_repository(@original_repository.id)
    AssignmentRepo.destroy_all
  end
  ...
```